plugins/package-managers/cargo/src/main/kotlin/Cargo.kt

```diff
@@ -266,7 +266,8 @@ private fun parseSourceArtifact(pkg: CargoMetadata.Package, hashes: Map<String,
 
     if (source == "registry+https://github.com/rust-lang/crates.io-index") {
         val url = "https://crates.io/api/v1/crates/${pkg.name}/${pkg.version}/download"
-        val hash = Hash.create(hashes[pkg.id].orEmpty())
+        val key = "${pkg.name} ${pkg.version} (${pkg.source})"
+        val hash = Hash.create(hashes[key].orEmpty())
```

Comment:
Commit message: Why not?

Reply:
I've explained it in the commit message now.